ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -57,111 +53,104 @@
-      "Second part of the prefix for persitent queries.";
 
   public static final String
       KSQL_TRANSIENT_QUERY_NAME_PREFIX_CONFIG = "ksql.transient.prefix";
-  public static final ConfigDef.Type
-      KSQL_TRANSIENT_QUERY_NAME_PREFIX_TYPE = ConfigDef.Type.STRING;
   public static final String
       KSQL_TRANSIENT_QUERY_NAME_PREFIX_DEFAULT = "transient_";
-  public static final ConfigDef.Importance
-      KSQL_TRANSIENT_QUERY_NAME_PREFIX_IMPORTANCE = ConfigDef.Importance.MEDIUM;
-  public static final String
-      KSQL_TRANSIENT_QUERY_NAME_PREFIX_DOC =
-      "Second part of the prefix for transient queries.";
 
   public static final String
       KSQL_TABLE_STATESTORE_NAME_SUFFIX_CONFIG = "ksql.statestore.suffix";
-  public static final ConfigDef.Type
-      KSQL_TABLE_STATESTORE_NAME_SUFFIX_TYPE = ConfigDef.Type.STRING;
-  public static final String
-      KSQL_TABLE_STATESTORE_NAME_SUFFIX_DEFAULT = "transient_";
-  public static final ConfigDef.Importance
-      KSQL_TABLE_STATESTORE_NAME_SUFFIX_IMPORTANCE = ConfigDef.Importance.MEDIUM;
   public static final String
-      KSQL_TABLE_STATESTORE_NAME_SUFFIX_DOC =
-      "Suffix for state store names in Tables.";
+      KSQL_TABLE_STATESTORE_NAME_SUFFIX_DEFAULT = "_ksql_statestore";
 
-  public int defaultSinkNumberOfPartitions = 4;
-  public short defaultSinkNumberOfReplications = 1;
+  private static int defaultSinkNumberOfPartitions = 4;
+  private static short defaultSinkNumberOfReplications = 1;
   // TODO: Find out the best default value.
-  public long defaultSinkWindowChangeLogAdditionalRetention = 1000000;
+  private static long defaultSinkWindowChangeLogAdditionalRetention = 1000000;
 
-  public String defaultAutoOffsetRestConfig = "latest";
-  public long defaultCommitIntervalMsConfig = 2000;
-  public long defaultCacheMaxBytesBufferingConfig = 10000000;
-  public int defaultNumberOfStreamsThreads = 4;
+  private static String defaultAutoOffsetRestConfig = "latest";
+  private static long defaultCommitIntervalMsConfig = 2000;
+  private static long defaultCacheMaxBytesBufferingConfig = 10000000;
+  private static int defaultNumberOfStreamsThreads = 4;
 
   Map<String, Object> ksqlConfigProps;
   Map<String, Object> ksqlStreamConfigProps;
 
-  private static final ConfigDef CONFIG_DEF = new ConfigDef();
+  private static final ConfigDef CONFIG_DEF;
+
+  static {
+    CONFIG_DEF = new ConfigDef()
+    .define(KSQL_SERVICE_ID_CONFIG,
+            ConfigDef.Type.STRING,
+            KSQL_SERVICE_ID_DEFAULT,
+            ConfigDef.Importance.MEDIUM,
+            "Indicates the ID of the ksql service. It will be used as prefix for all KSQL queries in "
+            + "this service.")
+
+    .define(KSQL_PERSISTENT_QUERY_NAME_PREFIX_CONFIG,
+            ConfigDef.Type.STRING,
+            KSQL_PERSISTENT_QUERY_NAME_PREFIX_DEFAULT,
+            ConfigDef.Importance.MEDIUM,
+            "Second part of the prefix for persitent queries. For instance if the prefix is transient_"
```

Comment:
It's the second part. The first part come from `KSQL_SERVICE_ID_CONFIG(ksql.service.id)`

Reply:
Ok. I am still not sure I follow. I think it would make sense for the example to include the full chain, so that it is clearer to the user how it all comes together at each stage.

Reply:
Added more details with an example.

Reply:
The doc string here as written is: 

> Second part of the prefix for persitent queries. For instance if the prefix is transient_query_ the query name will be ksql_query_1.

Did you really mean for the example prefix to be 'transient_query_'?